src/audio/asrc/asrc.h

```diff
@@ -165,4 +169,73 @@ struct ipc4_asrc_module_cfg {
 	uint32_t asrc_mode;
 } __packed __aligned(4);
 #endif
+
+/* Simple count value to prevent first delta timestamp
+ * from being input to low-pass filter.
+ */
+#define TS_STABLE_DIFF_COUNT	2
+
+/* Low pass filter coefficient for measured drift factor,
+ * The low pass function is y(n) = c1 * x(n) + c2 * y(n -1)
+ * coefficient c2 needs to be 1 - c1.
+ */
+#define COEF_C1		Q_CONVERT_FLOAT(0.01, 30)
+#define COEF_C2		Q_CONVERT_FLOAT(0.99, 30)
+
+typedef void (*asrc_proc_func)(struct processing_module *mod,
+			       const struct audio_stream *source,
+			       struct audio_stream *sink,
+			       int *consumed,
+			       int *produced);
+/* asrc component private data */
+struct comp_data {
+#if CONFIG_IPC_MAJOR_4
+	struct ipc4_asrc_module_cfg ipc_config;
+#else
+	struct ipc_config_asrc ipc_config;
+#endif
+	struct asrc_farrow *asrc_obj;	/* ASRC core data */
+	struct comp_dev *dai_dev;	/* Associated DAI component */
+	enum asrc_operation_mode mode;  /* Control for push or pull mode */
+	uint64_t ts;
+	uint32_t sink_rate;	/* Sample rate in Hz */
+	uint32_t source_rate;	/* Sample rate in Hz */
+	uint32_t sink_format;	/* For used PCM sample format */
+	uint32_t source_format;	/* For used PCM sample format */
+	uint32_t copy_count;	/* Count copy() operations  */
+	int32_t ts_prev;
+	int32_t sample_prev;
+	int32_t skew;		/* Rate factor in Q2.30 */
+	int32_t skew_min;
+	int32_t skew_max;
+	int ts_count;
+	int asrc_size;		/* ASRC object size */
+	int buf_size;		/* Samples buffer size */
+	int frames;		/* IO buffer length */
+	int source_frames;	/* Nominal # of frames to process at source */
+	int sink_frames;	/* Nominal # of frames to process at sink */
+	int source_frames_max;	/* Max # of frames to process at source */
+	int sink_frames_max;	/* Max # of frames to process at sink */
+	int data_shift;		/* Optional shift by 8 to process S24_4LE */
+	uint8_t *buf;		/* Samples buffer for input and output */
+	uint8_t *ibuf[PLATFORM_MAX_CHANNELS];	/* Input channels pointers */
+	uint8_t *obuf[PLATFORM_MAX_CHANNELS];	/* Output channels pointers */
+	bool track_drift;
+	asrc_proc_func asrc_func;		/* ASRC processing function */
+};
+
+int asrc_init(struct processing_module *mod);
+int asrc_dai_configure_timestamp(struct comp_data *cd);
+int asrc_dai_start_timestamp(struct comp_data *cd);
+int asrc_dai_stop_timestamp(struct comp_data *cd);
+#if CONFIG_IPC_MAJOR_4 && CONFIG_ZEPHYR_NATIVE_DRIVERS
+int asrc_dai_get_timestamp(struct comp_data *cd, struct dai_ts_data *tsd);
+#else
+int asrc_dai_get_timestamp(struct comp_data *cd, struct timestamp_data *tsd);
+#endif
+void update_buffer_format(struct comp_buffer *buf_c, struct comp_data *cd);
+void set_stream_params(struct comp_data *cd, struct sof_ipc_stream_params *params);
+extern const struct sof_uuid asrc_uuid;
```

Comment:
I did not add those extern, should no need for this.

Reply:
look at https://github.com/thesofproject/sof/pull/8161

Reply:
yes, src/asrc have different uuid for ipc3 and ipc4, seems this is the only way.